### PR TITLE
consider only required headers for aws signing request

### PR DESCRIPTION
**Pull request by kinarashah**
address https://github.com/rancher/rancher/pull/17983#issuecomment-462876922

### CODE REVIEW

File: pkg/httpproxy/sign.go [hunk 1]
```diff
@@ -235,3 +235,16 @@ type basic struct{}
 type digest struct{}
 
 type arbitrary struct{}
+
+func ignoreHeaders(req *http.Request) map[string]string {
+	data := map[string]string{}
+	for header, val := range req.Header {
+		if strings.EqualFold(header, ConnectionHeader) || strings.HasPrefix(strings.ToLower(header), ForwardHeaderPrefix) {
+			if len(val) > 0 {
+				data[header] = strings.Join(val, "")
```
Comment (alena1108) [inline pkg/httpproxy/sign.go]: why are we joining with an empty string?

Comment (kinarashah) [reply pkg/httpproxy/sign.go]: val is of type `[]string`

File: pkg/httpproxy/sign.go [hunk 2]
```diff
@@ -19,12 +19,15 @@ import (
 	"github.com/rancher/types/apis/core/v1"
 )
 
+const (
+	ConnectionHeader    = "connection"
```
Comment (alena1108) [inline pkg/httpproxy/sign.go]: @vincent99 can you think of any other headers that might need to get dropped before signing the request?

**Comment by alena1108** [comment]
LGTM. TF is unrelated. @kinarashah confirmed with @jgreat and @davidnuzik that the fix addresses the problem seen on aws setup.

**Pull request merged**